<a href="https://colab.research.google.com/github/gabikohen/Ejercicio-pr-ctico-sobre-rboles-de-clasificaci-n-cancer./blob/master/Ejercicio_pr%C3%A1ctico_sobre_%C3%A1rboles_de_clasificaci%C3%B3n%2Bcancer.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
## Mount Drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [8]:
## Import Libraries
import matplotlib.pyplot as plt
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import accuracy_score
from sklearn.ensemble import BaggingClassifier
from sklearn.ensemble import RandomForestClassifier


In [17]:
filename = "/content/drive/MyDrive/Colab Notebooks/Coding dojo/Introducion al Machine learning/Teoria/Clase 1/cancer.csv"
data = pd.read_csv(filename,index_col=0)

In [18]:
data.head()

diagnosis  radius_mean  texture_mean  perimeter_mean  area_mean  \
id                                                                         
842302           M        17.99         10.38          122.80     1001.0   
842517           M        20.57         17.77          132.90     1326.0   
84300903         M        19.69         21.25          130.00     1203.0   
84348301         M        11.42         20.38           77.58      386.1   
84358402         M        20.29         14.34          135.10     1297.0   

          smoothness_mean  compactness_mean  concavity_mean  \
id                                                            
842302            0.11840           0.27760          0.3001   
842517            0.08474           0.07864          0.0869   
84300903          0.10960           0.15990          0.1974   
84348301          0.14250           0.28390          0.2414   
84358402          0.10030           0.13280          0.1980   

          concave points_mean  symmetry_mean  ...  radius_worst  \
id                                            ...                 
842302                0.14710         0.2419  ...         25.38   
842517                0.07017         0.1812  ...         24.99   
84300903              0.12790         0.2069  ...         23.57   
84348301              0.10520         0.2597  ...         14.91   
84358402              0.10430         0.1809  ...         22.54   

          texture_worst  perimeter_worst  area_worst  smoothness_worst  \
id                                                                       
842302            17.33           184.60      2019.0            0.1622   
842517            23.41           158.80      1956.0            0.1238   
84300903          25.53           152.50      1709.0            0.1444   
84348301          26.50            98.87       567.7            0.2098   
84358402          16.67           152.20      1575.0            0.1374   

          compactness_worst  concavity_worst  concave points_worst  \
id                                                                   
842302               0.6656           0.7119                0.2654   
842517               0.1866           0.2416                0.1860   
84300903             0.4245           0.4504                0.2430   
84348301             0.8663           0.6869                0.2575   
84358402             0.2050           0.4000                0.1625   

          symmetry_worst  fractal_dimension_worst  
id                                                 
842302            0.4601                  0.11890  
842517            0.2750                  0.08902  
84300903          0.3613                  0.08758  
84348301          0.6638                  0.17300  
84358402          0.2364                  0.07678  

[5 rows x 31 columns]

In [19]:
data.tail()

diagnosis  radius_mean  texture_mean  perimeter_mean  area_mean  \
id                                                                       
926424         M        21.56         22.39          142.00     1479.0   
926682         M        20.13         28.25          131.20     1261.0   
926954         M        16.60         28.08          108.30      858.1   
927241         M        20.60         29.33          140.10     1265.0   
92751          B         7.76         24.54           47.92      181.0   

        smoothness_mean  compactness_mean  concavity_mean  \
id                                                          
926424          0.11100           0.11590         0.24390   
926682          0.09780           0.10340         0.14400   
926954          0.08455           0.10230         0.09251   
927241          0.11780           0.27700         0.35140   
92751           0.05263           0.04362         0.00000   

        concave points_mean  symmetry_mean  ...  radius_worst  texture_worst  \
id                                          ...                                
926424              0.13890         0.1726  ...        25.450          26.40   
926682              0.09791         0.1752  ...        23.690          38.25   
926954              0.05302         0.1590  ...        18.980          34.12   
927241              0.15200         0.2397  ...        25.740          39.42   
92751               0.00000         0.1587  ...         9.456          30.37   

        perimeter_worst  area_worst  smoothness_worst  compactness_worst  \
id                                                                         
926424           166.10      2027.0           0.14100            0.21130   
926682           155.00      1731.0           0.11660            0.19220   
926954           126.70      1124.0           0.11390            0.30940   
927241           184.60      1821.0           0.16500            0.86810   
92751             59.16       268.6           0.08996            0.06444   

        concavity_worst  concave points_worst  symmetry_worst  \
id                                                              
926424           0.4107                0.2216          0.2060   
926682           0.3215                0.1628          0.2572   
926954           0.3403                0.1418          0.2218   
927241           0.9387                0.2650          0.4087   
92751            0.0000                0.0000          0.2871   

        fractal_dimension_worst  
id                               
926424                  0.07115  
926682                  0.06637  
926954                  0.07820  
927241                  0.12400  
92751                   0.07039  

[5 rows x 31 columns]

In [10]:
data.duplicated().sum()

0

In [11]:
data.isna().sum().sum()

0

In [25]:
target = "diagnosis"
y = data[target]
X = data.drop(columns = target)

In [26]:
# train-test-split
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42)
X_train

radius_mean  texture_mean  perimeter_mean  area_mean  \
id                                                                
8913            12.890         13.12           81.89      515.9   
915691          13.400         20.52           88.64      556.7   
904689          12.960         18.29           84.18      525.2   
9110732         17.750         28.03          117.30      981.6   
881046502       20.580         22.14          134.70     1290.0   
...                ...           ...             ...        ...   
859711           8.888         14.64           58.79      244.0   
863031          11.640         18.33           75.17      412.5   
8910721         14.290         16.82           90.30      632.6   
908489          13.980         19.62           91.12      599.5   
862965          12.180         20.52           77.22      458.7   

           smoothness_mean  compactness_mean  concavity_mean  \
id                                                             
8913               0.06955           0.03729         0.02260   
915691             0.11060           0.14690         0.14450   
904689             0.07351           0.07899         0.04057   
9110732            0.09997           0.13140         0.16980   
881046502          0.09090           0.13480         0.16400   
...                    ...               ...             ...   
859711             0.09783           0.15310         0.08606   
863031             0.11420           0.10170         0.07070   
8910721            0.06429           0.02675         0.00725   
908489             0.10600           0.11330         0.11260   
862965             0.08013           0.04038         0.02383   

           concave points_mean  symmetry_mean  fractal_dimension_mean  ...  \
id                                                                     ...   
8913                   0.01171         0.1337                 0.05581  ...   
915691                 0.08172         0.2116                 0.07325  ...   
904689                 0.01883         0.1874                 0.05899  ...   
9110732                0.08293         0.1713                 0.05916  ...   
881046502              0.09561         0.1765                 0.05024  ...   
...                        ...            ...                     ...  ...   
859711                 0.02872         0.1902                 0.08980  ...   
863031                 0.03485         0.1801                 0.06520  ...   
8910721                0.00625         0.1508                 0.05376  ...   
908489                 0.06463         0.1669                 0.06544  ...   
862965                 0.01770         0.1739                 0.05677  ...   

           radius_worst  texture_worst  perimeter_worst  area_worst  \
id                                                                    
8913             13.620          15.54            87.40       577.0   
915691           16.410          29.66           113.30       844.4   
904689           14.130          24.61            96.31       621.9   
9110732          21.530          38.54           145.40      1437.0   
881046502        23.240          27.84           158.30      1656.0   
...                 ...            ...              ...         ...   
859711            9.733          15.67            62.56       284.4   
863031           13.140          29.26            85.51       521.7   
8910721          14.910          20.65            94.44       684.6   
908489           17.040          30.80           113.90       869.3   
862965           13.340          32.84            84.58       547.8   

           smoothness_worst  compactness_worst  concavity_worst  \
id                                                                
8913                0.09616            0.11470          0.11860   
915691              0.15740            0.38560          0.51060   
904689              0.09329            0.23180          0.16040   
9110732             0.14010     

In [30]:
# instanciar el modelo
dec_tree = DecisionTreeClassifier(random_state = 42)
# Entreno el modelo
dec_tree.fit(X_train, y_train)

# evaluo el rendimiento del modelo
train_score = dec_tree.score(X_train, y_train)
test_score = dec_tree.score(X_test, y_test)

# print
print(f'La precisión en el conjunto de entrenamiento es {train_score:.3f}.')
print(f'La precisión en el conjunto de prueba es {test_score:.3f}.')



La precisión en el conjunto de entrenamiento es 1.000.
La precisión en el conjunto de prueba es 0.951.


2) clasificador bagging y un

In [31]:
#  instacio el metodo
bagreg = BaggingClassifier(random_state = 42)

# entreno el metodo
bagreg.fit(X_train, y_train)

# predigo el medolo
bagreg.predict(X_test)
bagreg.predict(X_train)

# evuluo el rendimiento

bagreg_train_score = bagreg.score(X_train, y_train)
bagreg_test_score = bagreg.score(X_test, y_test)

# print
print(bagreg_train_score)
print(bagreg_test_score)



0.9929577464788732
0.951048951048951


3) clasificador de bosque aleatorio

para obtener la exactitud más alta posible en el conjunto de prue

In [40]:
#  instacio el metodo
rc= RandomForestClassifier(random_state = 42)

# entreno el metodo
rc.fit(X_train, y_train)

# predigo el medolo
test_preds=rc.predict(X_test)
train_preds=rc.predict(X_train)

# evuluo el rendimiento
train_score = accuracy_score(y_train, train_preds)
test_score = accuracy_score(y_test, test_preds)

# print
print(train_score)
print(test_score)

1.0
0.965034965034965


El mejor modelo predeterminado, según la precisión en el conjunto de prueba, es el RandomForestClassifier
- Training accuracy: 1.0
- Test accuracy: 0.965